src/libslic3r/PrintObject.cpp

```diff
@@ -2017,14 +2024,18 @@ void PrintObject::bridge_over_infill()
                     }
                 }
                 unsupported_area = closing(unsupported_area, float(SCALED_EPSILON));
-                // By expanding the lower layer solids, we avoid making bridges from the tiny internal overhangs that are (very likely) supported by previous layer solids
-                // NOTE that we cannot filter out polygons worth bridging by their area, because sometimes there is a very small internal island that will grow into large hole
-                lower_layer_solids = shrink(lower_layer_solids, 1 * spacing); // first remove thin regions that will not support anything
-                lower_layer_solids = expand(lower_layer_solids, (1 + 3) * spacing); // then expand back (opening), and further for parts supported by internal solids
-                // By shrinking the unsupported area, we avoid making bridges from narrow ensuring region along perimeters.
-                unsupported_area   = shrink(unsupported_area, 3 * spacing);
-                unsupported_area   = diff(unsupported_area, lower_layer_solids);
-
+                
+                // Orca:
+                // Lightning infill benefits from always having a bridge layer so don't filter out small unsupported areas. Also, don't filter small internal unsupported areas if the user has requested so.
+                if(!has_lightning_infill && !po->config().dont_filter_internal_bridges){
```

Comment:
Nice handle for the lightning_infill :)

Do you think it would be better to use an area threshold instead of a simple on/off option? I recall having a brief discussion with @Noisyfox previously about this part of the code. The issue we're facing is with 'thin but long' areas that are currently being filtered out, which isn’t ideal. The existing logic doesn't adequately address this. Adding a area threshold should be sufficient to handle these cases.
I would love to hear your thought as well.

Reply:
Open to do so, but will need guidance on how... Maybe if Noisyfox/you could also push to his PR?

Reply:
`are you thinking something like this code? 		
```
// Filter out contours with zero area or small area, contours with 2 points only.
        const double min_area_threshold = 0.01 * aoffset2 * aoffset2;
        remove_small(polygons_outer, min_area_threshold);
```

Used here: struct ExPolygonWithOffset
{
public:
    ExPolygonWithOffset(

Reason I didn't try it was because of the comment in the file "NOTE that we cannot filter out polygons worth bridging by their area, because sometimes there is a very small internal island that will grow into large hole".

Or are you thinking of something else?

Reply:
Sure thing.
I should be free to do some experiment after fixing last two issues for 1.9 beta.

Meanwhile here is the rough idea in my mind:
```
              // Orca: calc unsupported area without shrink&expand `lower_layer_solids` and `unsupported_area`
                auto unsupported_area_original   = diff(unsupported_area, lower_layer_solids);
                // By expanding the lower layer solids, we avoid making bridges from the tiny internal overhangs that are (very likely) supported by previous layer solids
                // NOTE that we cannot filter out polygons worth bridging by their area, because sometimes there is a very small internal island that will grow into large hole
                lower_layer_solids = shrink(lower_layer_solids, 1 * spacing); // first remove thin regions that will not support anything
                lower_layer_solids = expand(lower_layer_solids, (1 + 3) * spacing); // then expand back (opening), and further for parts supported by internal solids
                // By shrinking the unsupported area, we avoid making bridges from narrow ensuring region along perimeters.
                unsupported_area   = shrink(unsupported_area, 3 * spacing);
                unsupported_area   = diff(unsupported_area, lower_layer_solids);
                // Orca check threshold
                auto diff = diff(unsupported_area_original, unsupported_area);
                double area_diff = 0.0;
                for(auto p:diff)
                    area_diff += p.area()
                 // check with threshold in setting
                 if(area_diff > BRIDGE_THREASHOLD)
                      unsupported_area   = std::move(unsupported_area_original);
```

Reply:
Cool idea! I'll have a play with it as well tomorrow